src/Engine/Physics/Trigger.cpp

```diff
@@ -1,45 +1,39 @@
-#include "Trigger.hpp"
-
+#include <algorithm>
 #include <btBulletDynamicsCommon.h>
 #include "../Component/RigidBody.hpp"
+#include "Trigger.hpp"
+#include "TriggerObserver.hpp"
 
 namespace Physics {
-
-    Trigger::Trigger(const btTransform& transform) : btCollisionWorld::ContactResultCallback() {
+    Trigger::Trigger(const btTransform& transform) {
         trigger = new btCollisionObject();
         trigger->setWorldTransform(transform);
     }
 
-    // Called with each contact for our own processing. This is where we can
-    // test whether the given listeners are intersecting the trigger volume.
-    btScalar Trigger::addSingleResult(btManifoldPoint& cp,
-        const btCollisionObjectWrapper* colObj0, int partId0, int index0,
-        const btCollisionObjectWrapper* colObj1, int partId1, int index1) {
-
-        auto otherCollisionObject = (colObj0->getCollisionObject() == trigger) ? colObj1 : colObj0;
-        auto body = static_cast<btRigidBody*>(const_cast<btCollisionObject*>(otherCollisionObject->getCollisionObject()));
-        
-        // Call the corresponding callback
-        observers[body]();
-        return 0; // Was a planned purpose, but is not used.
-    }
-
     btCollisionObject* Trigger::GetCollisionObject() const {
         return trigger;
     }
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The function 'addSingleResult' is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)